lib/gs-plugin-loader.c

```diff
@@ -1729,9 +1767,54 @@ gs_plugin_loader_pending_apps_remove (GsPluginLoader *plugin_loader,
 		}
 
 	}
+	if (save_queue) {
+		g_clear_pointer (&locker, g_mutex_locker_free);
+		save_install_queue (plugin_loader);
+	}
 	g_idle_add (emit_pending_apps_idle, g_object_ref (plugin_loader));
 }
 
+static gboolean
+load_install_queue (GsPluginLoader *plugin_loader, GError **error)
+{
+	GsPluginLoaderPrivate *priv = gs_plugin_loader_get_instance_private (plugin_loader);
+	guint i;
+	g_autofree gchar *contents = NULL;
+	g_autofree gchar *file = NULL;
+	g_auto(GStrv) names = NULL;
+	g_autoptr(GMutexLocker) locker = NULL;
+
+	/* load from file */
+	file = g_build_filename (g_get_user_data_dir (),
+				 "gnome-software",
+				 "install-queue",
+				 NULL);
+	if (!g_file_test (file, G_FILE_TEST_EXISTS))
+		return TRUE;
+	g_debug ("loading install queue from %s", file);
+	if (!g_file_get_contents (file, &contents, NULL, error))
+		return FALSE;
+
+	locker = g_mutex_locker_new (&priv->pending_apps_mutex);
+	/* add each app-id */
+	names = g_strsplit (contents, "\n", 0);
+	for (i = 0; names[i]; i++) {
+		g_autoptr(GsApp) app = NULL;
+		const gchar *id = names[i];
+
+		if (strlen (id) == 0)
+			continue;
+		if (!as_utils_unique_id_valid (id))
+			continue;
+
+		app = gs_plugin_loader_app_create (plugin_loader, id);
+		gs_app_list_add (priv->pending_apps, app);
+		g_debug ("adding pending app %s", gs_app_get_unique_id (app));
```

Comment:
On the final file comparison, here is the debug statement after the action.

Reply:
Ah, I see it now. I can change that.